libs/feature/catalog/src/lib/organisations/service/organisations-from-metadata.service.ts

```diff
@@ -220,16 +205,15 @@ export class OrganisationsFromMetadataService
           (o) => o.name === metadataRecord.resourceContacts[0]?.organisation
         )[0]
 
-        if (org) {
-          const contactFromOrg = this.mapContactFromOrganisation(
-            org,
-            metadataRecord.contact
-          )
-          metadataRecord.contact = contactFromOrg
-          metadataRecord.resourceContacts = [
-            contactFromOrg, // FIXME: this should go into an organization field
-            ...metadataRecord.resourceContacts,
-          ]
+        if (
+          org &&
+          !metadataRecord.resourceContacts?.[0].logoUrl &&
+          org.logoUrl
+        ) {
+          const logoUrl = getAsUrl(`${org.logoUrl}`)
+          metadataRecord.resourceContacts[0].logoUrl = logoUrl // FIXME: this should go into an organization field
+          metadataRecord.resourceContacts[0].email ??= org.email
+          metadataRecord.contact.email ??= org.email
```

Comment:
The `if` statement is not correct for the email set.

I would suggest to keep your `mapContactFromOrganisation` and change the name to `hydrateContactWithOrg`, and do all the stuff in there.

I would also not touch the `record.contact`, you are guessing `record.contact` and `record.resourceContacts[0]` are the same but it might not be the case.

Note that the `group` strategy seems quite different as it adds new contacts within the metadata, which is a bit weird IMO, anyway, the group has the priority in that case so it's ok.